Source/WebKit/PlatformWPE.cmake

```
@@ -536,6 +536,10 @@ endif ()
 
 WEBKIT_BUILD_INSPECTOR_GRESOURCES(${WebInspectorUI_DERIVED_SOURCES_DIR} "inspector.gresource")
 
+add_custom_target(inspector_gresource_target
+    DEPENDS ${WebInspectorUI_DERIVED_SOURCES_DIR}/inspector.gresource
+)
+add_dependencies(WebKit inspector_gresource_target)
```

Why the `_target` suffix? We already know it's a target. Also, the `WebKit` target does not need `inspector.gresource` to be built, what you want is to add the `ALL` flag in the `add_custom_target()` call:

```suggestion
add_custom_target(inspector-resources
    DEPENDS ${WebInspectorUI_DERIVED_SOURCES_DIR}/inspector.gresource
    ALL
)
```

---

If you don't want to assume knowledge of how the `WEBKIT_BUILD_INSPECTOR_GRESOURCES()` macro works, you can move the `add_custom_target()`call _into_ the macro itself. Making the custom target depend on `ALL` works fine for both the WPE and GTK ports&mdash;in the case of GTK, the source dependency will pull the custom command earlier in the build process as a dependency for the `WebKit` target anyway (as intended).

---

Thanks for the `ALL`  suggestion. I just had to reorder it so `ALL` comes right after the target name. Patch updated.